src/freenas/usr/local/lib/middlewared_truenas/plugins/enclosure_/map.py

```diff
@@ -0,0 +1,108 @@
+from collections import namedtuple
+import re
+
+from middlewared.service import Service, private
+
+ProductMapping = namedtuple("ProductMapping", ["product_re", "mappings"])
+VersionMapping = namedtuple("VersionMapping", ["version_re", "slots"])
+MappingSlot = namedtuple("MappingSlot", ["num", "slot", "identify"])
+
+
+MAPPINGS = [
+    ProductMapping(re.compile(r"To Be Filled By O.E.M."), [
+        VersionMapping(re.compile(".*"), [
+            MappingSlot(0, 1, True),
+            MappingSlot(1, 0, True),
+        ]),
+    ]),
+    ProductMapping(re.compile(r"FREENAS-MINI-3.0-E\+?"), [
+        VersionMapping(re.compile(".*"), [
+            MappingSlot(0, 0, False),
+            MappingSlot(0, 1, False),
+            MappingSlot(0, 2, False),
+            MappingSlot(0, 3, False),
+            MappingSlot(0, 5, False),
+            MappingSlot(0, 4, False),
+        ]),
+    ]),
+]
+
+
+class EnclosureService(Service):
+    @private
+    async def map_enclosures(self, enclosures):
+        info = await self.middleware.call("system.info")
+        for product_mapping in MAPPINGS:
+            if product_mapping.product_re.match(info["system_product"]):
+                for version_mapping in product_mapping.mappings:
+                    if version_mapping.version_re.match(info["system_product_version"]):
```

Comment:
Is there any hardware you're seeing that will need the version?

Reply:
NEP says that mapping can vary from version to version

Reply:
Can you remind me which section?

Reply:
Ah, nevermind. its there just for future use and not actually used yet.